src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/TextBufferExtensions.cs

```diff
@@ -32,17 +32,6 @@ public static bool TryGetHostDocumentSyncVersion(this ITextBuffer textBuffer, ou
             return result;
         }
 
-        public static bool IsRazorLSPBuffer(this ITextBuffer textBuffer)
-        {
-            if (textBuffer == null)
-            {
-                throw new ArgumentNullException(nameof(textBuffer));
-            }
-
-            var matchesContentType = textBuffer.ContentType.IsOfType(RazorLSPConstants.RazorLSPContentTypeName);
-            return matchesContentType;
-        }
-
         public static void MakeEmptyEdit(this ITextBuffer textBuffer)
```

Comment:
Everything in this class is a somewhat "hack "to enable virtual document synchronization. After moving it over if you have any ideas on how to improve / make it less hack let me know.

Reply:
Sounds good. This is one part I haven't fully grokked yet, will definitely think about it. 

---
In reply to: [437130624](https://github.com/dotnet/aspnetcore-tooling/pull/1994#discussion_r437130624) [](ancestors = 437130624)